apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -15,12 +15,30 @@ function __construct() {
 		}
 		self::$initialized = true;
 
+		if( ! $this->is_full_site_editing_allowed() ) {
+			return;
+		}
+
 		add_action( 'init', array( $this, 'register_blocks' ), 100 );
 		add_action( 'init', array( $this, 'register_wp_template' ) );
 		add_action( 'rest_api_init', array( $this, 'allow_searching_for_templates' ) );
 		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_script_and_style' ), 100 );
 	}
 
+	/**
+	 * Returns true if the current site is not VIP
+	 * and we are running on a WordPress.com sandbox and proxied.
+	 * Or, if WordPress is in debug mode, for a local development environment.
+	 */
+	function is_full_site_editing_allowed() {
+		return (
+				function_exists( 'wpcom_is_proxied_request' ) && wpcom_is_proxied_request()
```

Comment:
Aside from the previous note, this will also prevent the plugin from loading in any other environment which doesn't have these functions and constants defined.